app/docker/services/imageService.js

```diff
@@ -138,6 +138,25 @@ angular.module('portainer.docker')
     return Image.tag({id: id, tag: imageConfig.tag, repo: imageConfig.repo}).$promise;
   };
 
+  service.downloadImages = function(images) {
+    var deferred = $q.defer();
+    var names = ImageHelper.getImagesNamesForDownload(images);
+    Image.download(names).$promise
+    .then(function success(data) {
+      var err = data.length > 0 && data[data.length - 1].hasOwnProperty('message');
```

Comment:
Is this error check really needed? Looks like it will check for an error in a response containing an array?